### PR TITLE
scenario_api_simulator port to ROS2

### PR DESCRIPTION
**Pull request by nnmm**
# Summary
This directly integrates the NPC simulator into the scenario_api_simulator package. We can thereby eliminate the awkwardness of a service call (see https://github.com/tier4/planning_simulator.iv.universe/pull/5#discussion_r522297262) and simply call the `getObject()` function on the object directly. This assumes no other nodes use the npc_simulator service.

# Changes in this PR
* Fixed a lot of unused and missing headers and packages.
* `ament_uncrustify` – that was a mistake since it makes the diff much larger. Sorry about this.
* Made `NPCRouteManager` (in this package) and `NPCSimulatorNode` (in its own package) an extension node – it receives a `rclcpp::Node&` in the constructor and extracts a logger and clock that it stores
* Same for the main node, `ScenarioAPISimulator`, except it receives an `rclcpp::Node::SharedPtr` for consistency with [`scenario_api_autoware`](https://github.com/tier4/scenario_runner.iv.universe/pull/12)
* Removed pointless `ros::spinOnce();` in timer callback – the node was already spinning, just before the timer callback.
* Changes to the `npc_simulator` build by @fred-apex-ai to ensure that transitive build dependencies (like `vehicle_info_util`) work correctly
* Made `npc_simulator::getObject` public and make `scenario_api_simulator` call it directly.
* Fixed name collision for `normalizeRadian` between `scenario_api_utils` and `npc_simulator`, and reduced the risk of further collisions by making `toMsg` and `calcDist2D` helper functions have internal linkage.

# TODOs for future PRs
* Do not use `std_msgs`
* We can remove the service in the NPC simulator if we go with this.
* Maybe move the `npc_simulator` into this package since it is now invalid to run it outside of this node

### CODE REVIEW

File: npc_simulator/CMakeLists.txt [hunk 1]
```diff
@@ -37,7 +37,6 @@ rosidl_generate_interfaces(${PROJECT_NAME}
 ament_export_dependencies(rosidl_default_runtime)
 
 ament_auto_add_library(${PROJECT_NAME}_node STATIC src/node.cpp)
-target_include_directories(${PROJECT_NAME}_node PRIVATE "${Boost_INCLUDE_DIR}")
```
Comment (fred-apex-ai) [inline npc_simulator/CMakeLists.txt:40]: Why did you remove this? I added it as you suggested it

File: api/scenario_api_simulator/package.xml [hunk 2]
```diff
@@ -22,6 +22,8 @@
   <depend>sensor_msgs</depend>
   <depend>std_msgs</depend>
   <depend>unique_identifier_msgs</depend>
+  <!-- TODO: This is not a dependency, only a workaround for build problems -->
```
Comment (fred-apex-ai) [inline api/scenario_api_simulator/package.xml]: what exactly is the problem with linking? I guess only `npc_simulator` should take care of it. Is it a problem with linking, perhaps the linker order because dependencies are not exported properly?

File: api/scenario_api_simulator/include/scenario_api_simulator/scenario_api_simulator.h [hunk 3]
```diff
@@ -105,7 +106,7 @@ class ScenarioAPISimulator
 private:
   rclcpp::Logger logger_;
   rclcpp::Clock::SharedPtr clock_;
-  rclcpp::Client < npc_simulator::srv::GetObject > ::SharedPtr client_;            //!< @brief private ros service client
+  std::shared_ptr < NPCSimulatorNode > npc_simulator_;
```
Comment (fred-apex-ai) [inline api/scenario_api_simulator/include/scenario_api_simulator/scenario_api_simulator.h]: If `npc_simulator_` is not a node any more, we should rename the class to `NPCSimulator`

File: api/scenario_api_simulator/CMakeLists.txt [hunk 4]
```diff
@@ -13,7 +13,7 @@ find_package(ament_cmake REQUIRED)
 find_package(ament_cmake_auto REQUIRED)
 ament_auto_find_build_dependencies()
 
-add_library(scenario_api_simulator SHARED
+add_library(scenario_api_simulator STATIC
```
Comment (fred-apex-ai) [inline api/scenario_api_simulator/CMakeLists.txt]: Why did you make it `STATIC`? This could lead to linker problems when the order is not quite right. But of course that should be fixed elsewhere

**Comment by nnmm** [comment]
@mitsudome-r This is now ready for merging and CI should pass after https://github.com/tier4/planning_simulator.iv.universe/pull/10/files is merged

**Review by fred-apex-ai — APPROVED** [review]

File: api/scenario_api_simulator/CMakeLists.txt (hunk 4 above)
Comment (nnmm) [reply api/scenario_api_simulator/CMakeLists.txt]: It has to be static as far as I understand because npc_simulator is static because vehicle_info_utils is static because that's the default.

File: api/scenario_api_simulator/include/scenario_api_simulator/scenario_api_simulator.h (hunk 3 above)
Comment (nnmm) [reply api/scenario_api_simulator/include/scenario_api_simulator/scenario_api_simulator.h]: Done

File: api/scenario_api_simulator/package.xml (hunk 2 above)
Comment (nnmm) [reply api/scenario_api_simulator/package.xml]: That line was from a previous commit, it's not in the current version of the code.

File: npc_simulator/CMakeLists.txt (hunk 1 above)
Comment (nnmm) [reply npc_simulator/CMakeLists.txt:40]: Added it back, it was probably because of experiments trying to get it to compile.
